Add a Wasm build for .NET 7

bricelam opened the pull request:
Part of dotnet/efcore#28999

ericsink commented:
Looks like the build of the native bits succeeded, but the attempt to build and run SQLitePCLRaw itself failed because the necessary tweaks have not yet been made to that repo.

The GH actions in these two repos have a bit of a "chicken and egg" problem.

bricelam commented:
Updated to flow new assets into the test job. Reverted moving the existing wasm libs (the publish job should take care of this since it clears the bin directory then adds the new libraries before pushing)
